src/test/java/com/flightstats/hub/metrics/StatsdReporterIntegrationTest.java

```diff
@@ -42,15 +43,20 @@ private static StatsdReporter provideStatsDHandlers() {
         return provider.get();
     }
 
+    private static void runAfterSocketInit(Runnable runnable) {
+        Executors.newScheduledThreadPool(1).schedule(runnable, 100, TimeUnit.MILLISECONDS);
```

Comment:
I'm surprised this works. The returned Executor is local to the the function scope and I'd think it'd get garbage collected and thus stop rescheduling. Maybe sufficiently reliable in a unit test though.

Reply:
ScheduledThreadPoolExecutor has an onShutdown() hook that ensures delayed tasks execute even after shutdown (by default).  Periodic tasks will be canceled though.